src/vm/eventpipe.cpp

```diff
@@ -421,8 +425,18 @@ void EventPipe::DisableInternal(EventPipeSessionID id, EventPipeProviderCallback
     if (s_pConfig == nullptr || !s_pConfig->Enabled())
         return;
 
+    // Get the specified session ID.
     EventPipeSession *pSession = nullptr;
-    if (!s_pSessions->Lookup(id, &pSession) || (pSession == nullptr))
+    uint64_t i = 0;
+    for (; i < MaxNumberOfSessions; ++i)
+    {
+        pSession = s_pSessions[i].Load();
+        if (pSession != nullptr && pSession->GetId() == id)
+            break;
+    }
+
+    // If the session was not found, then there is nothing else to do.
+    if (pSession == nullptr)
         return;
```

Comment:
The return statement will not execute if the last slot is non-empty but id mismatch, is that what you want?

Reply:
Look like we can reuse `EventPipe::GetSession` here